### PR TITLE
Using jwt mode if the server responded to an empty request with an "i…

### PR DESCRIPTION
**Pull request by askonev**
…nvalid token" error

### CODE REVIEW

File: config/StaticData.rb [hunk 1]
```diff
@@ -23,6 +23,16 @@ class StaticData
 
   EXCEPTION_FILES = JSON.load_file("#{Dir.pwd}/config/exception_file.json")
 
+  INVALID_TOKEN_ERROR = '-8'
+
+  def self.documentserver_jwt_exist?
+    ENV.include?('DOCUMENTSERVER_JWT')
```
Comment (ShockwaveNN) [inline config/StaticData.rb:29]: `ENV.key?(....)`

File: config/StaticData.rb [hunk 2]
```diff
@@ -23,6 +23,16 @@ class StaticData
...
+  end
+
+  def self.documentserver_jwt_empty?
+    ENV.fetch('DOCUMENTSERVER_JWT', '') != ''
```
Comment (ShockwaveNN) [inline config/StaticData.rb:33]:
```
ENV.fetch('DOCUMENTSERVER_JWT') != ''
```

And juding by name - it should check if key is empty

So correct check will be

```
ENV.fetch('DOCUMENTSERVER_JWT') == ''
```

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -26,11 +26,18 @@ def s3
...
+                       OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url,
+                                                                                     jwt_key: ENV.fetch('DOCUMENTSERVER_JWT'))
+                     elsif StaticData.jwt_data_exist?
+                       OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url,
+                                                                                     jwt_key: StaticData.get_jwt_key.strip)
+                     else
+                       raise 'JWT key is not defined'
+                     end
+                   else
                      OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url)
```
Comment (ShockwaveNN) [inline spec/spec_helper.rb:40]: I think this should be splitted not in one very complicated and multilevel `if` but in several different if with early exit

like this
```
def converter
return @converter if @converter

return @converter =  OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url) unless PretestsCheck.jwt_enable?

return @converter = OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url,
                                                                                     jwt_key: StaticData.get_jwt_key.strip) if StaticData.jwt_data_exist?
                                                                                     ....
end
```

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -26,11 +26,18 @@ def s3
 
 RSpec.configure do |config|
   def converter
-    @converter ||= if !StaticData::JWT_ENABLE
+    @converter ||= if PretestsCheck.jwt_enable?
+                     if StaticData.documentserver_jwt_exist? && StaticData.documentserver_jwt_empty?
```
Comment (ShockwaveNN) [inline spec/spec_helper.rb:30]: Maybe cobine output of both of those methods in 

```
def jwt_in_env?
  StaticData.documentserver_jwt_exist? && !StaticData.documentserver_jwt_empty?
end
```

and rename `documentserver_jwt_exist` `jwt_in_env?` and `documentserver_jwt_empty` to `jwt_in_env_empty?`

And rename `jwt_data_exist` to `jwt_in_config_file?`

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

File: config/StaticData.rb (hunk 2 above)
Comment (askonev) [reply config/StaticData.rb:33]: If the variable does not exist, an exception is raised.

https://apidock.com/ruby/Hash/fetch

So I use the default empty string to check the condition

Comment (ShockwaveNN) [reply config/StaticData.rb:33]: Fair enoght

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -26,12 +26,15 @@ def s3
...
   def converter
-    @converter ||= if !StaticData::JWT_ENABLE
-                     OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url)
-                   elsif StaticData.jwt_data_exist?
-                     OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url,
-                                                                                   jwt_key: StaticData.get_jwt_key.strip)
-                   end
+    return @converter if @converter
+
+    return @converter = OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url) unless PretestsCheck.jwt_enable?
```
Comment (ShockwaveNN) [inline spec/spec_helper.rb:31]: I think we can extract those long lines into three methods
```
def no_jwt_converter
OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url)
end

def jwt_from_env_converter
OnlyofficeDocumentserverConversionHelper::ConvertFileData.new(StaticData.documentserver_url,
                                                                               jwt_key: ENV.fetch('DOCUMENTSERVER_JWT'))
end

def jwt_from_file_converter
...
```

And simplify `def converter` method to
```
return @converter = no_jwt_converter unless PretestsCheck.jwt_enable? 
return @converter =  jwt_from_env_converter if StaticData...
...
```

And after that I think all rubocop warnings will be fixed

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

**Review by ShockwaveNN — APPROVED** [review]

**Review by l8556 — APPROVED** [review]

**Pull request merged**